File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for(int i = 0; i < nums.length; i++){
            int comp = target - nums[i];

            if(map.containsKey(comp)){
                int j = map.get(comp);

                return new int[] { Math.min(i, j), Math.max(i, j)};
            }

            map.put(nums[i], i);
        }

        return new int[] {};
    }
}

File: Data Structures & Algorithms/remove-element/submission-1.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {

        int currPos = 0;
        
        for(int i = 0; i < nums.size(); i++){
            if(nums[i] != val) nums[currPos++] = nums[i];
        }

        return currPos;
    }
};